lib/matplotlib/backend_tools.py

```diff
@@ -452,22 +453,16 @@ def add_figure(self):
         if self.figure not in self.views:
             self.views[self.figure] = cbook.Stack()
             self.positions[self.figure] = cbook.Stack()
+            self.home_views[self.figure] = WeakKeyDictionary()
             # Define Home
             self.push_current()
-            # Adding the clear method as axobserver, removes this burden from
-            # the backend
-            self.figure.add_axobserver(self.clear)
```

Comment:
@fariza 
Good point.  I'll add `update_home_views` as an axobserver.  Should I remove the ToolbarNavigation changes or leave those in?